crates/router/src/connector/payme/transformers.rs

```diff
@@ -1018,6 +1021,86 @@ impl TryFrom<types::RefundsResponseRouterData<api::Execute, PaymeRefundResponse>
     }
 }
 
+#[derive(Debug, Serialize)]
+pub struct PaymeVoidRequest {
+    sale_currency: enums::Currency,
+    payme_sale_id: String,
+    seller_payme_id: Secret<String>,
+    language: String,
+}
+
+impl
+    TryFrom<
+        &PaymeRouterData<
+            &types::RouterData<api::Void, types::PaymentsCancelData, types::PaymentsResponseData>,
+        >,
+    > for PaymeVoidRequest
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: &PaymeRouterData<
+            &types::RouterData<api::Void, types::PaymentsCancelData, types::PaymentsResponseData>,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let auth_type = PaymeAuthType::try_from(&item.router_data.connector_auth_type)?;
+        Ok(Self {
+            payme_sale_id: item.router_data.request.connector_transaction_id.clone(),
+            seller_payme_id: auth_type.seller_payme_id,
+            sale_currency: item.router_data.request.get_currency()?,
+            language: LANGUAGE.to_string(),
+        })
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct PaymeVoidResponse {
+    sale_status: SaleStatus,
+    payme_transaction_id: Option<String>,
+    status_error_code: Option<u32>,
+}
+
+impl TryFrom<types::PaymentsCancelResponseRouterData<PaymeVoidResponse>>
+    for types::PaymentsCancelRouterData
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::PaymentsCancelResponseRouterData<PaymeVoidResponse>,
+    ) -> Result<Self, Self::Error> {
+        let status = enums::AttemptStatus::from(item.response.sale_status.clone());
+        let response = if is_payment_failure(status) {
+            let payme_response = &item.response;
+            let status_error_code = payme_response.status_error_code.ok_or(
```

Comment:
Don't validate the field and throw error. Instead if the value is None default error_code, error_message to NO_ERROR_CODE & NO_ERROR_MESSAGE